dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -463,8 +425,18 @@ ModelMulti::evalMode(const double & t, const vector<double> &y, const vector<dou
   std::copy(y.begin(), y.end(), yLocal_);
   std::copy(yp.begin(), yp.end(), ypLocal_);
   std::copy(z.begin(), z.end(), zLocal_);
-  for (unsigned int i = 0; i < subModels_.size(); ++i)
-    subModels_[i]->evalModeSub(t);
+
+  modeChange_ = false;
+  modeChangeType_t modeChangeType = modeChangeType_t::NO_MODE;
```

Comment:
No need for an intermediate variable here, we can use directly modeChangeType_

Reply:
Is it expected to not go back to NO_MODE when modeChange == false?

Reply:
There are two mechanisms there, which explains why you need to have this intermediate modeChangeType and why modeChangeType_ doesn't go back to NO_MODE when modeChange is false.

modeChange_ is used to denote a change in the mode during a call of evalMode (especially in the loop in DYNSolverImpl in evalZMode that looks for changes in z or in mode). 
The modeChangeType_ is set for the overall time step, which means that at a certain call to evalMode, you can have no modeChange (modeChange_ = false) but in the complete time step you had an ALGEBRAIC_MODE_CHANGE. You then need this info at the end of the time step on the solver side to determine which treatment you have to do (reinit or not for example). It also explains why the reinitialization to NO_MODE for modeChangeType_ is done outside by the solvers themselves.

Nevertheless, I will add a comment to explain this.